File: wrappers/dotnet/src/BbsSignatures/Models/BlindSignRequest.cs
using System;

namespace BbsSignatures
{
    /// <summary>
    /// A request to create a BBS signature that features blinded/commited messages
    /// </summary>
    public class BlindSignRequest
    {
        public BlindSignRequest(BlsKeyPair secretKey, BbsKey publicKey, byte[] commitment, IndexedMessage[] messages)
        {
            KeyPair = secretKey ?? throw new ArgumentNullException(nameof(secretKey));
            Key = publicKey ?? throw new ArgumentNullException(nameof(publicKey));
            Commitment = commitment ?? throw new ArgumentNullException(nameof(commitment));
            Messages = messages ?? throw new ArgumentNullException(nameof(messages));
        }

        /// <summary>
        /// The secret key of the signer
        /// </summary>
        public BlsKeyPair KeyPair { get; set; }

        /// <summary>
        /// The public key of the signer
        /// </summary>
        public BbsKey Key { get; set; }

        /// <summary>
        /// The resulting commitment of the blinded messages to sign
        /// </summary>
        public byte[] Commitment { get; set; }

        /// <summary>
        /// The known messages to sign
        /// </summary>
        public IndexedMessage[] Messages { get; set; }
    }
}

File: wrappers/python/ursa_bbs_signatures/models/BlindSignRequest.py
from typing import List
from .keys import BbsKey, BlsKeyPair
from .IndexedMessage import IndexedMessage


class BlindSignRequest:
    def __init__(
        self,
        secret_key: BlsKeyPair,
        public_key: BbsKey,
        commitment: bytes,
        messages: List[IndexedMessage],
    ) -> None:
        """A request to create a BBS signature that features blinded/committed messages

        Args:
            secret_key: The secret key of the signer
            public_key: The public key of the signer
            commitment: The resulting commitment of the blinded messages to sign
            messages: The known messages to sign
        """
        self.secret_key = secret_key
        self.public_key = public_key
        self.commitment = commitment
        self.messages = messages
